doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,396 @@
+COMMIT                           NAME             TYPE COMMITTED    SIZE
+8560235e7d854eae80aa03a33f8927eb /OHVoterData.csv file 1 second ago 153.8KiB
+```
+If you were to re-run the `put-file` command above,
+by default,
+the file would be appended to itself and listing the repo would look like this:
+```
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED     SIZE
+105aab526f064b58a351fe0783686c54 /OHVoterData.csv file 2 seconds ago 307.6KiB
+```
+
+In this case, any pipelines that use this repo for input will see an updated file that has double the data in it.
+It may also have an intermediate header row. 
+(See [Specifying Header/Footer](../cookbook/splitting.html?highlight=header#specifying-header-footer) for details on headers and footers in files.)
+This is Pachyderm's default behavior.
+What if you want to overwrite the files?
+
+### Overwriting files
+
+This is where the `-o` (or `--overwrite`) flag comes in handy.  It will, as you've probably guessed, overwrite the file, rather than append it.
+```
+$ pachctl put-file voterData master -f OHVoterData.csv --overwrite
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED    SIZE
+8560235e7d854eae80aa03a33f8927eb /OHVoterData.csv file 1 second ago 153.8KiB
+$ pachctl put-file voterData master -f OHVoterData.csv --overwrite
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED    SIZE
+4876f99951cc4ea9929a6a213554ced8 /OHVoterData.csv file 1 second ago 153.8KiB
+```
+### Deduplication
+Pachyderm will deduplicate data loaded into input repositories.
+If you were to load another file that hashed identically to "OHVoterData.csv",
+there would be one copy of the data in Pachyderm with two files' metadata pointing to it.
+This works even when using the append behavior above.
+If you were to put a file named OHVoterData2004.csv that was identical to that first put-file of OHVoterData.csv,
+and then update OHVoterData.csv as shown above,
+there would be two sets of bits in Pachyderm:
+
+- a set of bits that would be returned when asking for the old branch of OHVoterData.csv & OHVoterData2004.csv and
+- a set of bits that would be appended to that first set to assemble the new, master branch of OHVoterData.csv.
+
+This deduping happens with each file as it's added to Pachyderm.
+We don't do deduping within the file ("intrafile deduplication") because this system is built to work with any file type.
+
+In short, 
+Pachyderm is smart about keeping the minimum set of bits in the object store and assembling the version of the file you (or your code!) have asked for.
+
+### Use cases for large datasets in single files
+
+#### Splitting large files in Pachyderm
+
+Unlike a system like Git,
+which expects almost all of your files to be text, 
+Pachyderm does not do intra-file diffing because we work with any file type: 
```

Comment:
Git doesn't do intra-file diffing either, this is a common misconception, it does more or less the same thing we do, keeps a full copy of each version of the file and content addresses it so that if the same file is stored twice it can reference the same content. The big difference is that git has a lot of tooling around textual diffs such as the `git diff` command, which does textual diffs of files by default, whereas our `diff-file` only does inter-file diffing because it doesn't assume it's text. But in terms of storage our implementation is very similar to git's.